### ISSUE TITLE
Some key names lead to request paths that do not comply to RFC3986, leading to failing requests in S3Mock

### ISSUE THREAD

**Issue by togise** [adobe/S3Mock#1686]
```java
GetObjectRequest getObjectRequest = GetObjectRequest.builder()
                    .bucket(bucketName)
                    .key(key)
                    .build();

s3Client.getObject(getObjectRequest, ResponseTransformer.toInputStream());
```
Is throwing an S3Exception with a http status code 400 instead of NoSuchKeyException when s3mock is called.
The last version of the s3Mock it works with is 3.2.0

**Comment by afranken** [comment]
@togise there is an integration test that tests this exact behavior, and it's running just fine:
https://github.com/adobe/S3Mock/blob/main/integration-tests/src/test/kotlin/com/adobe/testing/s3mock/its/ErrorResponsesV2IT.kt#L28

Not sure what's wrong in your code, you should look at the S3Mock logs, preferably in debug.

**Comment by afranken** [comment]
I should add: the integration tests are executed on every commit, releases are not possible if they do not pass.

**Comment by togise** [comment]
So I tried the key in your test `NoSuchKey.json` and it passed, but if I used my key which was `/no/file/exists` it failed.

**Comment by afranken** [comment]
found the problem.
Spring Boot 3.2 updated Jetty from 11 to 12.
Jetty 12 enforces paths according to this RFC: https://datatracker.ietf.org/doc/html/rfc3986#section-3.3
When you request a key `/test` in a bucket `test-bucket`, we receive a request `/test-bucket//test` which is not RFC compliant.
S3 however allows keys to be prefixed with a `/`.

For now, the behaviour in Jetty is configurable, and I'll release a fix in the next days that will allow you to continue using these keys.

**Comment by afranken** [comment]
For the record, the exception thrown by Jetty is:

```
org.eclipse.jetty.http.BadMessageException: 400: Ambiguous URI path separator
	at org.eclipse.jetty.server.internal.HttpConnection$HttpStreamOverHTTP1.headerComplete(HttpConnection.java:1186) ~[jetty-server-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.server.internal.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:952) ~[jetty-server-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1249) ~[jetty-http-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1545) ~[jetty-http-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.server.internal.HttpConnection.parseRequestBuffer(HttpConnection.java:567) ~[jetty-server-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.server.internal.HttpConnection.onFillable(HttpConnection.java:390) ~[jetty-server-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:322) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.ssl.SslConnection$SslEndPoint.onFillable(SslConnection.java:574) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:390) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:150) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:99) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53) ~[jetty-io-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:478) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:441) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:293) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.run(AdaptiveExecutionStrategy.java:201) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:410) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:971) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.doRunJob(QueuedThreadPool.java:1201) ~[jetty-util-12.0.6.jar:12.0.6]
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1156) ~[jetty-util-12.0.6.jar:12.0.6]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

**Comment by afranken** [comment]
@togise I just released [3.5.2](https://github.com/adobe/S3Mock/releases/tag/3.5.2) fixing this issue.

**Comment by afranken** [comment]
updating Spring Boot to 3.2.5 breaks URI handling yet again:
https://github.com/spring-projects/spring-boot/issues/40250
https://github.com/jetty/jetty.project/issues/11495
fixing this in #1785 again...
